Ensembl/ensembl-variation
# Create registry files from multiple hosts and from select species

This script uses the database configuration found in `/nfs/production/flicek/ensembl/variation/servers.txt` in order to create a registry file. Another config file may be used with argument `--server_file`.

The script supports creating a registry file with variation, core and funcgen databases hosted from different servers:
```bash
perl generate_reg_file.pl --species mouse,human --release 108 -v v1,v3 -c st1,st1b
```

The shortcuts used are defined in `server_file` and should be the same shortcuts that are currently used in Codon.

| Argument | Default value | Default value description |
| ------ | -- | -- |
| `--variation` | `v1-w,v3-w` | var-prod-1 and 3 (write permissions) |
| `--core`        | `v1,v3` | var-prod-1 and 3 (read-only) |
| `--funcgen` | `v1,v3` | var-prod-1 and 3 (read-only) |

If the same database exists in multiple servers, only the information from one of the servers will be randomly returned. It's random because we use hashes to store the database configuration. This could be improved.

### Major changes

* Updated the default path of `server_file`
* Updated list of species
* Replaced argument `--servers` with `--variation`, `--core` and `--funcgen` to explicitly configure the servers for each database
* Added argument `--species` to select one or multiple species
* Added argument `--dbname_var` to customise the name of the variation database

### Testing

1. Create registry with write access to databases in v1 and v3
```bash
perl generate_reg_file.pl --species mouse,human --release 108 > ensembl.registry
```

2. Create registry with write access to custom variation database in v1 and v3
    * Check if script raises an error when the custom database name is not found
```bash
perl generate_reg_file.pl --species mouse --release 108 --dbname_var user_mus_musculus_variation_108_3_test > ensembl.registry
```

3. Create registry with read-only access to databases in v1 and v3
```bash
perl generate_reg_file.pl --species mouse,human --release 108 -v v1,v3 > ensembl.registry
```

4. Create registry with core database from staging 1, 2 and 3
```bash
perl generate_reg_file.pl --species mouse,human --release 108 -c st1,st2,st3 > ensembl.registry
```
